app/src/main/java/ch/epfl/sweng/SDP/auth/Account.java

```diff
@@ -0,0 +1,338 @@
+     */
+    void checkIfAccountNameIsFree(final String newName)
+            throws IllegalArgumentException, DatabaseException {
+        if (newName == null) {
+            throw new IllegalArgumentException("Username must not be null");
+        }
+        usersRef.orderByChild("username").equalTo(newName)
+                .addListenerForSingleValueEvent(new ValueEventListener() {
+
+                    @Override
+                    public void onDataChange(@NonNull DataSnapshot snapshot) {
+                        if (snapshot.exists()) {
+                            throw new IllegalArgumentException("Username already taken.");
+                        }
+                    }
+
+                    @Override
+                    public void onCancelled(@NonNull DatabaseError databaseError) {
+                        throw databaseError.toException();
+                    }
+                });
+    }
+
+    /**
+     * Updates username to newName.
+     *
+     * @param newName new username
+     * @throws IllegalArgumentException if username not available anymore
+     * @throws DatabaseException if problems with Firebase
+     */
+    public void updateUsername(final String newName)
+            throws IllegalArgumentException, DatabaseException {
+        checkIfAccountNameIsFree(newName);
+        DatabaseReferenceBuilder builder = new DatabaseReferenceBuilder(usersRef);
+        builder.addChildren(userId + ".username").build()
+                .setValue(newName, createCompletionListener());
+        username = newName;
+        localDbHandler.saveAccount(this);
+    }
+
+    /**
+     * Method that allowes one to change trophies.
+     *
+     * @param change modifier of trophies
+     * @throws DatabaseException in case write to database fails
+     */
+    public void changeTrophies(final int change) throws DatabaseException {
+        DatabaseReferenceBuilder builder = new DatabaseReferenceBuilder(usersRef);
+        builder.addChildren(userId + ".trophies").build().addListenerForSingleValueEvent(
+                new ValueEventListener() {
+                    @Override
+                    public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
+                        Long value = dataSnapshot.getValue(Long.class);
+                        if (value != null) {
+                            final int newTrophies = Math.max(0, value.intValue() + change);
+                            DatabaseReferenceBuilder trophiesBuilder = new DatabaseReferenceBuilder(
+                                    usersRef);
+                            trophiesBuilder.addChildren(userId + ".trophies").build()
+                                    .setValue(newTrophies, createCompletionListener());
+                            trophies = newTrophies;
+                            
+                            updateCurrentLeague();
+
+                            localDbHandler.saveAccount(instance);
```

Comment:
I don't know if I understood correctly, but wasn't the point of storing the account in the local database, that values can instantly be updated? If you implement the localDBHandler like this, won't the values only change after Firebase was updated? What is the point of having account in the local DB then?